data_steward/constants/tools/create_combined_backup_dataset.py

```diff
@@ -0,0 +1,208 @@
+FOREIGN_KEYS_FIELDS = [
+    'visit_occurrence_id', 'location_id', 'care_site_id', 'provider_id',
+    'visit_detail_id'
+]
+RDR_TABLES_TO_COPY = ['person']
+EHR_TABLES_TO_COPY = ['death']
+DOMAIN_TABLES = list(
+    set(cdm.tables_to_map()) - set(RDR_TABLES_TO_COPY + EHR_TABLES_TO_COPY))
+TABLES_TO_PROCESS = RDR_TABLES_TO_COPY + EHR_TABLES_TO_COPY + DOMAIN_TABLES
+LEFT_JOIN = JINJA_ENV.from_string("""
+LEFT JOIN
+(
+SELECT *
+FROM (
+SELECT
+*,
+row_number() OVER (PARTITION BY {{prefix}}.{{field}}, {{prefix}}.src_hpo_id ) 
+AS row_num
+FROM `{{dataset_id}}.{{table}}` AS {{prefix}}
+)
+WHERE row_num = 1
+) {{prefix}}  ON t.{{field}} = {{prefix}}.src_{{field}}
+AND m.src_dataset_id = {{prefix}}.src_dataset_id
+""")
+
+JOIN_VISIT = JINJA_ENV.from_string("""
+JOIN
+(
+SELECT *
+FROM (
+SELECT
+*,
+row_number() OVER (PARTITION BY {{prefix}}.{{field}}, {{prefix}}.src_hpo_id ) 
+AS row_num
+FROM `{{dataset_id}}.{{table}}` AS {{prefix}}
+)
+WHERE row_num = 1
+) {{prefix}}  ON t.{{field}} = {{prefix}}.src_{{field}}
+AND m.src_dataset_id = {{prefix}}.src_dataset_id
+""")
+
+LEFT_JOIN_PERSON = JINJA_ENV.from_string("""
+LEFT JOIN
+(
+SELECT *
+FROM (
+SELECT
+*,
+row_number() OVER (PARTITION BY {{prefix}}.{{field}}, {{prefix}}.src_hpo_id ) 
+AS row_num
+FROM `{{dataset_id}}.{{table}}` AS {{prefix}}
+)
+WHERE row_num = 1
+) {{prefix}}  ON t.{{field}} = {{prefix}}.src_{{field}}
+""")
+
+EHR_CONSENT_QUERY = JINJA_ENV.from_string("""
+WITH ordered_response AS
+(SELECT
+person_id,
+value_source_concept_id,
+observation_datetime,
+ROW_NUMBER() OVER(PARTITION BY person_id ORDER BY observation_datetime DESC, 
+value_source_concept_id ASC) AS rn
+FROM `{{dataset_id}}.observation`
+WHERE observation_source_value = '{{source_value_ehr_consent}}')
+SELECT person_id
+FROM ordered_response
+WHERE rn = 1
+AND value_source_concept_id = {{concept_id_consent_permission_yes}}
+""")
+
+COPY_RDR_QUERY = JINJA_ENV.from_string(
+    """SELECT * FROM `{{rdr_dataset_id}}.{{table}}`""")
+
+COPY_EHR_QUERY = JINJA_ENV.from_string("""
+SELECT * FROM `{{ehr_dataset_id}}.{{table}}` AS t
+WHERE EXISTS
+   (SELECT 1 FROM `{{combined_dataset_id}}.{{ehr_consent_table_id}}` AS c
+    WHERE t.person_id = c.person_id)
+""")
+
+MAPPING_QUERY = JINJA_ENV.from_string("""
+SELECT DISTINCT
+'{{rdr_dataset_id}}'  AS src_dataset_id,
+{{domain_table}}_id  AS src_{{domain_table}}_id,
+'rdr' as src_hpo_id,
+{{domain_table}}_id + {{mapping_constant}}  AS {{domain_table}}_id,
+'{{domain_table}}' as src_table_id
+FROM `{{rdr_dataset_id}}.{{domain_table}}`
+
+UNION ALL
+
+SELECT DISTINCT
+'{{ehr_dataset_id}}'  AS src_dataset_id,
+t.{{domain_table}}_id AS src_{{domain_table}}_id,
+v.src_hpo_id AS src_hpo_id,
+t.{{domain_table}}_id  AS {{domain_table}}_id,
+'{{domain_table}}' as src_table_id
+FROM `{{ehr_dataset_id}}.{{domain_table}}` AS t
+JOIN `{{ehr_dataset_id}}._mapping_{{domain_table}}` AS v 
+ON t.{{domain_table}}_id = v.{{domain_table}}_id
+{% if {{person_id_flag}} -%}
```

Comment:
I think this will fix the failing unit test.
```suggestion
{% if person_id_flag %}
```